src/sparkx/Particle.py

```diff
@@ -267,130 +243,131 @@ def __initialize_from_array(self,input_format,particle_array):
-                "ncoll_": 12,
-                "form_time_": 13,
-                "xsecfac_": 14,
-                "proc_id_origin_": 15,
-                "proc_type_origin_": 16,
-                "t_last_coll_": 17,
-                "pdg_mother1_": 18,
-                "pdg_mother2_": 19,
-                "weight_": 20,
+                 "t_": [0,0],
+                "x_": [1,1],
+                "y_": [2,2],
+                "z_": [3,3],
+                "mass_": [4,4],
+                "E_": [5,5],
+                "px_": [6,6],
+                "py_": [7,7],
+                "pz_": [8,8],
+                "pdg_": [9,9],
+                "ID_": [11,10],
+                "charge_": [12,11],
+                "ncoll_": [13,12],
+                "form_time_": [14,11],
+                "xsecfac_": [15,14],
+                "proc_id_origin_": [16,15],
+                "proc_type_origin_": [17,16],
+                "t_last_coll_": [18,17],
+                "pdg_mother1_": [19,18],
+                "pdg_mother2_": [20,19],
+                "weight_": [24,20],
             },
             "JETSCAPE": {
-                "ID_": 0,
-                "pdg_": 1,
-                "status_": 2,
-                "E_": 3,
-                "px_": 4,
-                "py_": 5,
-                "pz_": 6,
+                "ID_": [11,0],
+                "pdg_": [9,1],
+                "status_": [21,2],
+                "E_": [5,3],
+                "px_": [6,4],
+                "py_": [7,5],
+                "pz_": [8,6],
             },
         }
         if (input_format in attribute_mapping):
             if (len(particle_array) == len(attribute_mapping[input_format]) or (input_format in ["Oscar2013Extended","Oscar2013Extended_IC"]\
                  and len(particle_array) <=  len(attribute_mapping[input_format])\
                     and len(particle_array) >=  len(attribute_mapping[input_format])-2)):
                 for attribute, index in attribute_mapping[input_format].items():
-                    if len(particle_array)<index+1:
-                        setattr(self,attribute,None)
+                    if len(particle_array)<index[1]+1:
                         continue
                     # Type casting for specific attributes
                     if attribute in ["t_", "x_", "y_", "z_", "mass_", "E_", "px_", "py_", "pz_", "form_time_", "xsecfac_", "t_last_coll_", "weight_"]:
-                        setattr(self, attribute, float(particle_array[index]))
-                    elif attribute in ["pdg_", "ID_", "charge_", "ncoll_", "proc_id_origin_", "proc_type_origin_", "pdg_mother1_", "pdg_mother2_", "baryon_number_", "strangeness_", "status_"]:
-                        setattr(self, attribute, int(particle_array[index]))
+                        self.data_[index[0]] = float(particle_array[index[1]])
+                    elif attribute in ["pdg_", "ID_", "ncoll_", "proc_id_origin_", "proc_type_origin_", "pdg_mother1_", "pdg_mother2_", "status_"]:
```

Comment:
Why is this still present? I thought that everything is now stored as `float`. Then we don't need the casts.

Reply:
That is a good point indeed. I added it to ensure that only only ints, even though in the format of a float, are stored. Imagine the user is a madman, feeding a file with 2.5 charge - how should we act? Allow the 2.5 charge, or emulate old behaviour by casting it to 2.0?

Reply:
Ok, then maybe add a comment there for other people who might stumble across this.

Reply:
Good idea, I will add the comment.